src/Poly/Monomial.hs

```diff
@@ -30,6 +31,9 @@ import Data.List qualified as L
 import Data.Maybe
 import Data.Vector.Storable qualified as VS
 import Data.Singletons
+#if MIN_VERSION_singletons(3,0,0)
+import GHC.TypeLits.Singletons
+#endif
```

Comment:
Is it actually used here?

Reply:
Yes, this module makes use of the `SingI` instance for `Symbol`